extras/foreman_protector/dnf/foreman-protector.py

```diff
@@ -0,0 +1,77 @@
+import dnf
+import dnf.exceptions
+from dnfpluginscore import _, logger
+
+import configparser
+
+class ForemanProtector(dnf.Plugin):
+    name = 'foreman-protector'
+    config_name = 'foreman-protector'
+
+    def __init__(self,base,cli):
+        self.base = base
+        self.cli = cli
+
+    def config(self):
+        global fileurl
+
+        try:
+             parser = self.read_config(self.base.conf)
+        except Exception as e:
+            raise dnf.exceptions.Error(_("Parsing file failed: {}").format(str(e)))
+
+        if parser.has_section('main'):
+            fileurl = parser.get('main', 'whitelist')
+        else:
+            raise dnf.exceptions.Error(_('Incorrect plugin configuration!'))
+
+    def _load_whitelist(self):
+        package_whitelist = set()
+        try:
+            if fileurl:
+                llfile = open(fileurl, 'r')
+                for line in llfile.readlines():
+                    if line.startswith('#') or line.strip() == '':
+                        continue
+
+                    package_whitelist.add(line.rstrip().lower())
```

Comment:
Not sure about `.lower()`. Package names are case sensitive.

Reply:
removed!